src/libraries/System.Private.CoreLib/src/System/Int128.cs

```diff
@@ -742,12 +742,19 @@ public static (Int128 Quotient, Int128 Remainder) DivRem(Int128 left, Int128 rig
 
         /// <inheritdoc cref="IBinaryInteger{TSelf}.LeadingZeroCount(TSelf)" />
         public static Int128 LeadingZeroCount(Int128 value)
+            => value.LeadingZeroCount();
+
+        /// <summary>Computes the number of leading zero bits in this value.</summary>
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        private int LeadingZeroCount()
         {
+            Int128 value = this;
+
```

Comment:
```suggestion
```
Can we fold the private method into the public method? I do not see the reason for the split.

Reply:
The public method returns Int128, but there are other methods on these types that want an int; seemed unnecessary to roundtrip through Int128 to get it.

Reply:
> The public method returns Int128

That is a weird design. Was it intentional?

Reply:
The public methods have to account for being used in a generic context, where you often expect or require `T in, T out` and where type changes can lead to additional work or complexity.

This then also matches how most hardware behaves when their is direct support for the operation over a given `T`.